### PR TITLE
WIP - DYN-5703 - Accept Excel Nodes

### PR DESCRIPTION
**Pull request by nmglob**
### Purpose

This PR is to make Excel nodes work on the NET6 solution

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Release Notes

Net6 Excel nodes should now be working

### Reviewers

@mjkkirschner

### CODE REVIEW

File: src/Libraries/DSOffice/DSOffice.csproj [hunk 1]
```diff
@@ -10,8 +10,8 @@
     <AssemblyName>DSOffice</AssemblyName>
     <DocumentationFile>$(OutputPath)\$(UICulture)\DSOffice.xml</DocumentationFile>
   </PropertyGroup>
-    <ItemGroup Condition=" '$(TargetFramework)' == 'net48' ">
-        <PackageReference Include="Microsoft.Office.Interop.Excel" Version="15.0.4795.1000">
+    <ItemGroup>
+        <PackageReference Include="Microsoft.Office.Interop.Excel" Version="15.0.4795.1001">
```
Comment (mjkkirschner) [inline src/Libraries/DSOffice/DSOffice.csproj:14]: any idea what the difference is between these packages?

File: src/Libraries/DSOffice/CompatibilityMarshal.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using System.Security;
+using System;
+
+/// <summary>
+/// Can't use Marshall on NET6 so this class replaces that functionality in order to keep the Excel.cs file (Excel nodes) working
+/// </summary>
+public static class CompatibilityMarshal
+{
+    internal const String OLEAUT32 = "oleaut32.dll";
+    internal const String OLE32 = "ole32.dll";
```
Comment (aparajit-pratap) [inline src/Libraries/DSOffice/CompatibilityMarshal.cs:13]: These can be private

File: src/Libraries/DSOffice/CompatibilityMarshal.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#if NET5_0_OR_GREATER
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Security;
+using System;
+
+/// <summary>
+/// Can't use Marshall on NET6 so this class replaces that functionality in order to keep the Excel.cs file (Excel nodes) working
+/// </summary>
+public static class CompatibilityMarshal
```
Comment (aparajit-pratap) [inline src/Libraries/DSOffice/CompatibilityMarshal.cs:10]: I think we can make this internal for now.

File: src/Libraries/DSOffice/CompatibilityMarshal.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+{
+    internal const String OLEAUT32 = "oleaut32.dll";
+    internal const String OLE32 = "ole32.dll";
+
+    /// <summary>
+    /// Works like Marshal.GetActiveObject on Net6
+    /// </summary>
+    [SecurityCritical]  // auto-generated_required
+    public static Object GetActiveObject(String progID)
```
Comment (aparajit-pratap) [inline src/Libraries/DSOffice/CompatibilityMarshal.cs:19]: internal

File: src/Libraries/DSOffice/DSOffice.csproj (hunk 1 above)
Comment (nmglob) [reply src/Libraries/DSOffice/DSOffice.csproj:14]: 1000 (2016) can use Office 2013 Excel interop
1001 (2022) can use for Excel 2013/2016/2019 COM interop
1001 also supports net standard 2

**Pull request merged**